### PR TITLE
[RCCA-17513] Fix group mapping errors in `confluent iam rbac role-binding` commands

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- Fix a bug preventing role bindings with an SSO group mapping principal from displaying in `confluent iam rbac role-binding list`
- Fix a bug causing an incorrect error message to be displayed in `confluent iam rbac role-binding create` and `confluent iam rbac role-binding delete` when the principal is an SSO group mapping

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
We weren't handling group mappings in `conflulent iam rbac role-binding` commands, so `list` omitted role bindings for group mappings, and `create|delete` treated group mapping IDs as user IDs and subsequently produced an error while trying to display the result, even though the underlying create/delete succeeded.

References
----------

Test & Review
-------------
I added a group mapping role binding to the test backend and updated the tests.
I also switched `pool-` to `group-` throughout the tests.

### CODE REVIEW

File: internal/iam/command_rbac_role_binding.go [hunk 1]
```diff
@@ -400,7 +400,7 @@ func (c *roleBindingCommand) displayCCloudCreateAndDeleteOutput(cmd *cobra.Comma
 	principalType := presource.LookupType(userId)
 
 	var fields []string
-	if principalType == presource.ServiceAccount || principalType == presource.IdentityPool {
+	if principalType == presource.ServiceAccount || principalType == presource.IdentityPool || principalType == presource.SsoGroupMapping {
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding.go:403]: https://pkg.go.dev/slices#Contains

File: internal/iam/command_rbac_role_binding_list.go [hunk 2]
```diff
@@ -491,6 +510,10 @@ func (c *roleBindingCommand) ccloudListRolePrincipals(cmd *cobra.Command, listRo
 			row.Name = name
 			list.Add(row)
 		}
+		if name, ok := groupMappingToNameMap[principal]; ok {
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding_list.go:513]: Unfortunate that we need to keep adding to this `for` loop every time we introduce a new resource.

File: test/fixtures/output/iam/rbac/role-binding/list-user-orgadmin-cloud.golden [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
-------------------+-----------------+-----------------------
-  User:pool-12345 | identity-pool   |                       
-  User:sa-12345   | service-account |                       
-  User:u-11aaa    |                 | u-11aaa@confluent.io  
+     Principal    |         Name          |        Email          
+------------------+-----------------------+-----------------------
+  User:group-abc  | another-group-mapping |                       
+  User:pool-12345 | identity-pool         |                       
+  User:sa-12345   | service-account       |                       
+  User:u-11aaa    |                       | u-11aaa@confluent.io
```
Comment (brianstrauch) [inline test/fixtures/output/iam/rbac/role-binding/list-user-orgadmin-cloud.golden:6]: What about combining the "Name" and "Email" column in v4?

Comment (sgagniere) [reply test/fixtures/output/iam/rbac/role-binding/list-user-orgadmin-cloud.golden:6]: That's an option. The other option is that we can just display the user's name here; I'm not sure why we currently aren't.

Comment (brianstrauch) [reply test/fixtures/output/iam/rbac/role-binding/list-user-orgadmin-cloud.golden:6]: Oh, I don't think that field was added until a year or so ago. We should definitely add that, but removing the email might cause problems.

File: internal/iam/command_rbac_role_binding.go [hunk 4]
```diff
@@ -54,6 +55,8 @@ var (
 
 	literalPatternType  = "LITERAL"
 	prefixedPatternType = "PREFIXED"
+
+	nonUserPrincipalTypes = []string{presource.ServiceAccount, presource.IdentityPool, presource.SsoGroupMapping}
```
Comment (brianstrauch) [inline internal/iam/command_rbac_role_binding.go:59]: Let's define this in the function it's used in? No need for it to be global in this package

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**